esp/services/ws_topology/ws_topologyService.cpp

```diff
@@ -1397,7 +1397,13 @@ bool CWsTopologyEx::onTpClusterInfo(IEspContext &context, IEspTpClusterInfoReque
         Owned<IRemoteConnection> conn = querySDS().connect("/Status/Servers/", myProcessSession(),RTM_SUB,SDS_LOCK_TIMEOUT);
         if (conn)
         {
+#ifndef _CONTAINERIZED
             Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(req.getName());
+#else
+            CConstWUClusterInfoArray clusters;
+            getContainerWUClusterInfo(clusters);
+            IConstWUClusterInfo* clusterInfo = queryContainerWUClusterInfoByName(req.getName(), clusters);
```

Comment:
Isn't getContainerWUClusterInfo building a list of all targets here (creating N CContainerWUClusterInfo objects),
only for all except <=1 to be thrown away after 1 is selected by this new queryContainerWUClusterInfoByName function?

It would be better to explicitly create the one you wanted (after checking it's a valid target), than create all, then discard all bar 1.

Reply:
Make sense